docs/developer-guide/ksqldb-rest-api/streaming-endpoint.md

```diff
@@ -0,0 +1,147 @@
+to consider using the Java client rather than using this API directly.
+
+These endpoints are only available when using HTTP 2.
+
+### Executing pull or push queries
+
+The request method will be a POST.
+
+Requests will be made to the endpoint "/query-stream".
+
+The body of the request is a JSON object UTF-8 encoded as text, containing the arguments for the
+operation (newlines have been added here for the sake of clarity but the real JSON must not contain
+ unescaped newlines)
+
+````
+{
+"sql": "select * from foo", <----- the SQL of the query to execute
+"properties": {             <----- Optional properties for the query
+    "prop1": "val1",
+    "prop2": "val2"
+   }
+}
+
+````
+
+The endpoint produces responses with two possible content types `application/json` and
+`application/vnd.ksqlapi.delimited.v1`. To choose one or the other you should set the `Accept`
+header in the request. The default is `application/vnd.ksqlapi.delimited.v1`.
+
+In the case of a successful query, if the content type is `application/vnd.ksqlapi.delimited.v1`
+then the results are returned as a header JSON object followed by zero or more JSON arrays
+delimited by newlines. Newline delimited formats are easy to parse by clients, and don't require
+a streaming JSON parser on the client in the case that intermediate results need to be output.
+
+````
+{
+"queryId", "xyz123",                          <---- unique ID of the query, used when terminating the query
+"columnNames":["col", "col2", "col3"],        <---- the names of the columns
+"columnTypes":["BIGINT", "STRING", "BOOLEAN"] <---- The types of the columns
+}
+````
+
+Followed by zero or more JSON arrays:
+
+````
+[123, "blah", true]
+[432, "foo", true]
+[765, "whatever", false]
+````
+
+If you prefer to receive the entire response as valid JSON then you can request
+content type `application/json`. In which case you will receive the results, as a single JSON
+array, for example: (newlines have been added for clarity, the response body
+won't contain newlines).
+
+````
+[
+{
+"queryId", "xyz123",                          <---- unique ID of the query, used when terminating the query
+"columnNames":["col", "col2", "col3"],        <---- the names of the columns
+"columnTypes":["BIGINT", "STRING", "BOOLEAN"] <---- The types of the columns
+},
+[123, "blah", true]
+[432, "foo", true]
+[765, "whatever", false]
+]
+````
+
+### Terminating queries
+
+Push queries can be explicitly terminated by the client by making a request to this endpoint
+
+The request method will be a POST.
+
+Requests will be made to the endpoint "/close-query"
+
+The body of the request is a JSON object UTF-8 encoded as text, containing the id of the 
+query to close. (newlines have been added here for the sake of clarity but the real JSON must not
+contain newlines)
+
+````
+{
+"queryId": "xyz123", <----- the ID of the query to terminate
+}
+
+````
+ 
+### Streaming inserts

```

Comment:
Perhaps replace with "Inserting rows into an existing stream" to clarify the use case?

Regardless of whether we change the subheading, we should clarify that this endpoint only allows insertion of rows into ksqlDB streams and not tables. (I see the comment in the example specifies "KSQL stream" but I think this is worth calling out explicitly in the text as well.)